### PR TITLE
feat: add stepreport example

### PR DESCRIPTION
**Pull request by cody82**
- Add a step report example
- Update documentation
- Remove return_on_error

### CODE REVIEW

**Comment by cody82** [comment]
I noticed the example yaml has unneeded connectors/auxiliaries in it, will remove those.

File: examples/test_stepreport/stepreport_example.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        super().setUp()
+        self.assertTrue(True, msg="Check my device is ready")
+        # additional data to include in the step-report
+        self.step_report.header["Version_device"] = "2022-1234"
+
+    def test_run(self):
+        """In this case the default test_run method is overridden and
+        instead of calling test_run from RemoteTest class the following
+        code is called.
+        Here, the test pass at the 3rd attempt out of 5. The setup and
```
Comment (Pog3k) [inline examples/test_stepreport/stepreport_example.py:50]: wrong copy paste?

Comment (cody82) [reply examples/test_stepreport/stepreport_example.py:50]: Yes, will fix it.

File: docs/advanced_usage/step_report.rst [hunk 2]
```diff
@@ -24,24 +24,26 @@ Usage Examples
...
 
-        # assert
+        # Simple assert
         self.assertTrue(device_on, msg="Check my device is ready")
 
-        # assert fail but continue on error
-        # test is set to failed if assertion does not succeed
-        self.step_report.continue_on_error = True
-        self.assertFalse(device_on, msg="Some check")
+        with self.subTest("Non critical checks"):
```
Comment (Pog3k) [inline docs/advanced_usage/step_report.rst:34]: Where dies the string "Non critical checks" appear ? 
can't see it in the step report

Comment (cody82) [reply docs/advanced_usage/step_report.rst:34]: Will only appear in the log output if it fails, not in the step report:

test_run (stepreport_example.StepReportTest-1-1) [Non critical checks]

FAIL: Traceback (most recent call last):
  File "\kiso-testing\examples\test_stepreport\stepreport_example.py", line 59, in test_run
    self.assertFalse(device_on, msg="Some check")
AssertionError: True is not false : Some check

**Review by Pog3k — APPROVED** [review]

**Review by sebastianpfischer — APPROVED** [review]

File: examples/test_stepreport/stepreport_example.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+        # Custom message is reset there
+        self.assertAlmostEqual(voltage, 4, delta=1, msg="Check voltage device")
+
+        logging.info(f"I HAVE RUN 0.1.1 for tag {self.tag}!")
+
+
+@pykiso.define_test_parameters(
+    suite_id=1,
+    case_id=2,
+)
```
Comment (sebclrsn) [inline examples/test_stepreport/stepreport_example.py:74]:
```suggestion
@pykiso.define_test_parameters(suite_id=1, case_id=2)
```

File: examples/test_stepreport/stepreport_example.py [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+class TableTest(pykiso.BasicTest):
+    """This test shows multiple tables in the step report"""
+
+    def setUp(self):
+        """Set header information and check setup conditions"""
+        super().setUp()
```
Comment (sebclrsn) [inline examples/test_stepreport/stepreport_example.py:80]: The super call is useless

File: examples/test_stepreport/stepreport_example.py [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+        # additional data to include in the step-report
+        self.step_report.header["Version_device"] = "2022-1234"
+
+    def test_run(self):
+        """In this test we specify custom tables to group the steps."""
+        logging.info(
+            f"--------------- RUN: {self.test_suite_id}, {self.test_case_id} ---------------"
+        )
```
Comment (sebclrsn) [inline examples/test_stepreport/stepreport_example.py:88]: This isn't necessary anymore as we already log the test information in nice frames

File: examples/test_stepreport/stepreport_example.py [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+# SPDX-License-Identifier: EPL-2.0
+##########################################################################
+
+"""
+test_suite_1
+************
+:module: test_suite_1
+:synopsis: Basic example on how to write and configure test suite and
+case on ITF side in order to communicate with the device under test
+using TestApp.
```
Comment (sebclrsn) [inline examples/test_stepreport/stepreport_example.py:16]:
```suggestion
:module: stepreport_example

:synopsis: Basic example showing the generation of
    an HTML step-report and its possible manipulation
    within the test.
    
    Run this example with:
    pykiso -c examples/step_report.yaml --step-report
```

File: examples/test_stepreport/stepreport_example.py [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+import logging
+from itertools import cycle
+
+import pykiso
+
+
+@pykiso.define_test_parameters(
+    suite_id=1,
+    case_id=1,
+)
```
Comment (sebclrsn) [inline examples/test_stepreport/stepreport_example.py:29]:
```suggestion
@pykiso.define_test_parameters(suite_id=1, case_id=1)
```

File: examples/test_stepreport/stepreport_example.py [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+************
+:module: test_suite_1
+:synopsis: Basic example on how to write and configure test suite and
+case on ITF side in order to communicate with the device under test
+using TestApp.
+"""
+
+import importlib
+import logging
+from itertools import cycle
```
Comment (sebclrsn) [inline examples/test_stepreport/stepreport_example.py:21]: Seems not to be used

**Review by sebclrsn — CHANGES_REQUESTED** [review]

**Pull request merged**